src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -27,31 +29,33 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
         {
             services.AddLogging();
 
+            services.AddMemoryCache();
+
             //
             // Add required services
             services.TryAddSingleton<IFeatureDefinitionProvider, ConfigurationFeatureDefinitionProvider>();
 
             services.AddSingleton(sp => new FeatureManager(
                 sp.GetRequiredService<IFeatureDefinitionProvider>(),
-                sp.GetRequiredService<IEnumerable<IFeatureFilterMetadata>>(),
-                sp.GetRequiredService<IEnumerable<ISessionManager>>(),
-                sp.GetRequiredService<ILoggerFactory>(),
-                sp.GetRequiredService<IOptions<FeatureManagementOptions>>(),
-                sp.GetRequiredService<IOptions<TargetingEvaluationOptions>>())
+                sp.GetRequiredService<IOptions<FeatureManagementOptions>>().Value)
             {
-                Configuration = sp.GetService<IConfiguration>(),
-                TargetingContextAccessor = sp.GetService<ITargetingContextAccessor>(),
+                FeatureFilters = sp.GetRequiredService<IEnumerable<IFeatureFilterMetadata>>(),
+                SessionManagers = sp.GetRequiredService<IEnumerable<ISessionManager>>(),
                 TelemetryPublishers = sp.GetService<IOptions<FeatureManagementOptions>>()?.Value.TelemetryPublisherFactories?
                     .Select(factory => factory(sp))
-                    .ToList()
+                    .ToList() ??
+                    Enumerable.Empty<ITelemetryPublisher>(),
+                Cache = sp.GetService<IMemoryCache>(),
+                Logger = sp.GetService<ILoggerFactory>().CreateLogger<FeatureManager>(),
```

Comment:
```suggestion
                Logger = sp.GetRequiredService<ILoggerFactory>().CreateLogger<FeatureManager>(),
```

We ensure logging is there before this. So we can require it.